### PR TITLE
Update Application list pages to Bootstrap 4

### PR DESCRIPTION
**Pull request by jsnshrmn**
## Description
- update application list to bootstrap 4
- drop jquery  2.2.4 and 3.4.1 in favor of 3.5.1
- reworks application list filter forms and button controls to be more consistent internally and with the rest of the site

## Rationale
Bootstrap 3, jquery 2.2.4, and jquery 3.4.1 are EOL

## Phabricator Ticket
https://phabricator.wikimedia.org/T293984

## How Has This Been Tested?
- lots of manual visiual checks on the application list, emulating multiple devices
- quick checks of js-driven behaviors on views where jquery was updated, such as suggest and send_partner

## Screenshots of your changes (if appropriate):
The desktop view is very much as expected, the mobile view is serviceable but not perfect in my opinion. The sizing of the filter inputs is difficult to control perfectly because they are inserted via javascript by DAL.
image

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

File: TWLight/applications/templates/applications/application_list.html [hunk 1]
```diff
@@ -1,229 +1,246 @@
...
+            {% comment %}Translator: This describes the status of a group of applications.{% endcomment %}
+            {% trans "Up for renewal" %}
+          </a>
+        </div>
       </div>
     </div>
-  </div>
 
-  <h1>{{ title }}</h1>
+    <h1>{{ title }}</h1>
```
Comment (suecarmol) [inline TWLight/applications/templates/applications/application_list.html:63]: Being a bit nitpicky here, but I think this header looks better with a `my-3` class

**Review by suecarmol — CHANGES_REQUESTED** [review]
I just have one very minor change request before we can merge this.

File: TWLight/applications/templates/applications/application_list.html (hunk 1 above)
Comment (jsnshrmn) [reply TWLight/applications/templates/applications/application_list.html:63]: Agreed, this looks much better!

**Review by suecarmol — APPROVED** [review]
Thanks for working on this and for updating the version of JQuery

**Pull request merged**